demisto_sdk/commands/common/docker/dockerhub_client.py

```diff
@@ -0,0 +1,382 @@
+import os
+from datetime import datetime, timedelta
+from enum import Enum
+from functools import lru_cache
+from typing import Any, Dict, List, Optional
+
+import dateparser
+import requests
+from packaging.version import InvalidVersion, Version
+from requests.exceptions import ConnectionError, HTTPError, Timeout
+
+from demisto_sdk.commands.common.handlers.xsoar_handler import JSONDecodeError
+from demisto_sdk.commands.common.logger import logger
+from demisto_sdk.commands.common.tools import retry
+
+DOCKERHUB_USER = "DOCKERHUB_USER"
+DOCKERHUB_PASSWORD = "DOCKERHUB_PASSWORD"
+DEFAULT_REPOSITORY = "demisto"
+
+
+class DockerHubAuthScope(str, Enum):
+    PULL = "pull"  # Grants read-only access to the repository, allowing you to pull images.
+    PUSH = "push"  # Grants write access to the repository, allowing you to push images.
+    DELETE = "delete"  # Grants permission to delete images from the repository.
+    REPOSITORY = "repository"  # Grants full access to the repository, including both pull and push permissions.
```

Comment:
Maybe move to constants?

Reply:
I think it would be better here as its related to dockerhub authentication options